components/tooltip/script.js

```diff
@@ -61,12 +63,24 @@ class Tooltip extends HTMLElement {
     hide () {
         this.style.display = 'none';
         this.visible = false;
+        // reset the current positioning as the page might get
+        // resized untill the next time the tooltip is displayed
+        this.fixedSides = [];

```

Comment:
There are the orientations (top, left, bottom, right) that we've tried already. We check all possible positions subsequently until we either find one that fixes the hidden tooltip or we run out of options. We need to track which orientations we've tries so far so that we don't try the same position multiple times or run over all them indefinitely.

The name is unclear and confusing, would appreciate suggestions.

Reply:
checkedOrientations ?

Reply:
Yes! A lot better, thanks.